### PR TITLE
Add ButterTextbox widget

### PR DESCRIPTION
**Pull request by humphd**
https://webmademovies.lighthouseapp.com/projects/65733-popcorn-maker/tickets/1165-create-a-textbox-widget-that-selects-on-click

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/bb3bc1687c26996/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://butyr.org:8000/592d9bee6beda61/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/bb3bc1687c26996/output.txt

Total script time: 0.33 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/592d9bee6beda61/output.txt

Total script time: 0.59 mins

Server is running on http://pr289.butyr.org/templates/test.html

File: src/ui/widget/textbox.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    input.addEventListener( "focus", __highlight, false );
+    input.addEventListener( "mouseup", __ignoreMouseUp, false );
+  }
+
+  return function( input ){
+    if( !(input && input.text) ){
+      throw "Expected an input element of type text";
+    }
+
+    input.addEventListener(
```
Comment (jbuck) [inline src/ui/widget/textbox.js]: Can you make this the regular style of addEventListener?

```
input.addEventListener( "blur", function( e ) {
  __addListeners( e.target );
}, false );
```

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/8150ec19cb0efbc/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/8150ec19cb0efbc/output.txt

Total script time: 0.53 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @jbuck received. Current queue size: 0

Live output at: http://butyr.org:8000/dc8c48e3160f56d/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/dc8c48e3160f56d/output.txt

Total script time: 1.13 mins

**Pull request merged**